src/rust/catnap/queue.rs

```diff
@@ -252,7 +221,7 @@ impl CatnapQueue {
                     }
                 },
                 Err(e) => {
-                    socket.rollback();
+                    self.socket.borrow_mut().rollback();
```

Comment:
Why should we keep some rollback logic on the queue level?

Reply:
Because I didn't want to move the yielder into the socket level, so I had to handle the cancel error there in the queue level.

Reply:
I understand, but then what is the point to have the responsibility of driving the state machine spread across the socket and queue level? After all, this is what motived this PR initially. I understand that the state machine is something independent of the transport layer (ie: Linux Sockets vs Shared Memory Rings), so for me it makes sense to have all that in the highets level possible. Eventually we want the queue mechanism be shared as much as possible.

Reply:
Yes, I think we should move the yielder into the socket level but maybe in another PR?

Reply:
> Yes, I think we should move the yielder into the socket level but maybe in another PR?

Sure we can.